### ISSUE TITLE
mysql_status: Update CheckMK 2.3 fails

### ISSUE THREAD

**Issue by TGion** [kuhn-ruess/Checkmk-Checks#76]
We are trying to update CheckMK from 2.2.0p25.cre to 2.3.
The update fails because of deprecated API calls:

```
01/06 Legacy check plugins...
Ignoring outdated plugin 'mysql.status': Format no longer supported -- this API is deprecated!
 
You can abort the update process (A) and try to fix the incompatibilities or continue the update (c).
```

Is this issue known? Are there any workarounds? 
Thanks in advance for your support!

**Comment by Bastian-Kuhn** [comment]
Hello,

There is no workaround, this plugin needs to be rewritten for the new Checkmk 2.3 API.

2.3 removed the 1.x APIs, deprecated the one introduced in 2.0 and introduced a whole new, but better supported Check API.

**Comment by TGion** [comment]
Hey Bastian,

thanks for your swift reply.
Will you consider the rewrite to the new API in the near future (or ever)?

**Comment by Bastian-Kuhn** [comment]
Hi Tobias,
as soon one of our consumers orders us.
Can't tell you how soon that happens.

**Comment by moschlar** [comment]
Hi everyone - I took the liberty to migrate to the old new API in #80.

@Bastian-Kuhn Please let me know what you need in order to incorporate this from a licensing standpoint. I'm happy to grant all rights.

**Comment by Bastian-Kuhn** [comment]
@moschlar Thank you very much, will look into it. From the Standpoint of License, that is the one we use: https://github.com/kuhn-ruess/Checkmk-Checks?tab=MIT-1-ov-file#readme (MIT)

**Comment by Tiippex** [comment]
Hey, thanks for the update. I have tested the new version with CheckMK 2.3.0p13 Enterprise.

The perf o meter shows me an exception: Exception: 'PerfDataTuple' object is not subscriptable
<img width="1435" alt="image" src="https://github.com/user-attachments/assets/acc16ae6-8e0d-44b4-b8bd-bd13c6490b60">

Thanks for your work!

**Comment by moschlar** [comment]
@Tiippex Darn, sorry - I have just tested it with 2.2 since I have a bunch of other plugins preventing the upgrade... Will look into it!

**Comment by moschlar** [comment]
@Tiippex A fix is in #87!
